common/src/main/java/com/microsoft/identity/common/internal/platform/DevicePopManager.java

```diff
@@ -1076,8 +1089,22 @@ private KeyPair generateNewRsaKeyPair(@NonNull final Context context,
                                 e
                         );
                         tryImport = false;
+
+                        if (null != e.getCause() && e.getCause().getClass().getSimpleName().equals(STRONG_BOX_UNAVAILABLE_EXCEPTION)) {
+                            // On some devices (notably, Huawei Mate 9 Pro), StrongBox errors are
+                            // the cause of the surfaced SecureKeyImportUnavailableException.
+                            tryStrongBox = false;
```

Comment:
Do we want a log similar to the above block here?

Reply:
Logging via e07f16c3a30df0e752d531fc831ce301a450e62f